src/js/filter.js

```diff
@@ -0,0 +1,27 @@
+/* global document, window */
+
+// can set default to certain filters
+const onFilterChange = (markerGroup, citiesToMarkers, data) => () => {
+  const checked = document.querySelectorAll(".scope-filter :checked");
+  const selected = [...checked].map((option) => option.value)[0];
+  console.log(selected);
+  Object.entries(citiesToMarkers).forEach(([cityState, marker]) => {
+    console.log("in here");
+    if (data[cityState]["report_magnitude"].includes(selected)) {
+      marker.addTo(markerGroup);
+    } else {
+      marker.removeFrom(markerGroup);
+    }
+  });
+};
+
+const setUpFilter = (markerGroup, citiesToMarkers, data) => {
+  filter = document.querySelector(".scope-filter");
+
+  filter.addEventListener(
+    "click",
```

Comment:
I wonder if there's a more specific event we can listen to, specifically when the values selected for the filter change. There's a non-trivial performance cost to running `onFilterChange` because it has to iterate over every single city we have  with your call to `.forEach()`. 

That has "time complexity" of O(n) -- look up what time complexity means if you haven't learned about it yet. It's an important way to think about how performance scales.